apps/io-sign-backoffice-app/src/app/api/api-keys/_lib/api-key.ts

```diff
@@ -124,9 +83,138 @@ async function deleteApimSubscription(subscriptionId: string) {
   }
 }
 
+async function apiKeyExists(institutionId: string, displayName: string) {
+  try {
+    const apiKeys = await getApiKeys(institutionId, { displayName });
+    if (apiKeys.length !== 0) {
+      throw new ApiKeyAlreadyExistsError(
+        "such name already exists for the institution"
+      );
+    }
+  } catch (e) {
+    throw e instanceof ApiKeyAlreadyExistsError
+      ? e
+      : new Error("unable to create the API key", { cause: e });
+  }
+}
+
+async function insertApiKey(apiKey: ApiKey) {
+  try {
+    await getCosmosContainer().items.create(apiKey);
+  } catch (e) {
+    await deleteApimSubscription(apiKey.id);
```

Comment:
Is this some kind of rollback? Maybe it would be better to handle this in the caller than as a side effet of insertApiKey
```suggestion
    await deleteApimSubscription(apiKey.id);
```